apps/hash-ai-worker-ts/src/activities/flow-activities/research-entities-action/infer-facts-from-web-page-worker-agent.ts

```diff
@@ -511,82 +463,77 @@ export const inferEntitiesFromWebPageWorkerAgent = async (params: {
-              linkEntityTypeIds: inferringLinkEntitiesOfTypeIds,
+              // entityTypeIds: inferringEntitiesOfTypeIds,
+              // linkEntityTypeIds: inferringLinkEntitiesOfTypeIds,
             } = toolCallInput;
 
-            if (
-              "expectedNumberOfEntities" in toolCallInput &&
-              toolCallInput.expectedNumberOfEntities < 1
-            ) {
-              return {
-                ...toolCall,
-                output: dedent(`
-                  You provided an expected number of entities which is less than 1. You must provide
-                  a positive integer as the expected number of entities to infer.
-                `),
-                isError: true,
-              };
-            }
-
-            const validEntityTypeIds = input.entityTypes.map(({ $id }) => $id);
-
-            const invalidEntityTypeIds = inferringEntitiesOfTypeIds.filter(
-              (entityTypeId) => !validEntityTypeIds.includes(entityTypeId),
-            );
-
-            const validLinkEntityTypeIds = input.linkEntityTypes?.map(
-              ({ $id }) => $id,
-            );
-
-            const invalidLinkEntityTypeIds =
-              inferringLinkEntitiesOfTypeIds?.filter(
-                (entityTypeId) =>
-                  !validLinkEntityTypeIds?.includes(entityTypeId),
-              ) ?? [];
-
-            if (
-              invalidEntityTypeIds.length > 0 ||
-              invalidLinkEntityTypeIds.length > 0
-            ) {
-              return {
-                ...toolCall,
-                output: dedent(`
-                  ${
-                    invalidEntityTypeIds.length > 0
-                      ? dedent(`
-                        You provided invalid entityTypeIds which don't correspond to the entity types
-                        which were initially provided: ${JSON.stringify(invalidEntityTypeIds)}
-
-                        The possible entity types you can submit are: ${JSON.stringify(
-                          validEntityTypeIds,
-                        )}
-                      `)
-                      : ""
-                  }
-                  ${
-                    invalidLinkEntityTypeIds.length > 0
-                      ? dedent(`
-                        You provided invalid linkEntityTypeIds which don't correspond to the link entity types
-                        which were initially provided: ${JSON.stringify(invalidLinkEntityTypeIds)}
-
-                        The possible link entity types you can submit are: ${JSON.stringify(
-                          validLinkEntityTypeIds,
-                        )}
-                      `)
-                      : ""
-                  }
-                `),
-                isError: true,
-              };
-            }
-
-            if (
-              "htmlContent" in toolCallInput
-                ? toolCallInput.htmlContent.length === 0
-                : toolCallInput.text.length === 0
-            ) {
+            // if (
```

Comment:
I'm assuming this is coming back in a future PR or being moved elsewhere

Reply:
There's some functionality which I've temporarily removed, but didn't want to remove the code yet for:
- `expectedNumberOfEntities`: allow the worker agent to set some expected number of entities, which is useful when it specifically knows how many entities facts should be inferred for (e.g. table data with 20 rows). This is not an argument the worker agent was consistently getting right, so didn't invest time adding this to the `inferFactsFromEntities` process, but may reintroduce later 
- `entityTypeIds` and `linkEntityTypeIds`: allows the worker agent to specify which types should be inferred from text. I removed this because we're already letting the coordinator agent decide this, and the more I think about it don't see the need for reintroducing this in the near future so have removed this in https://github.com/hashintel/hash/pull/4497/commits/2a898425a968df9b805382bad0032b42b0bf1870
- `validAt`: as we discussed previously, we need another approach for correctly determining when a fact is valid and this is likely not going to be addressed in the worker agent, or at least not be something that the worker agent is directly able to set. Meant to remove this but still spotted some commented out code which I removed in https://github.com/hashintel/hash/pull/4497/commits/18c5f7fb31eabffdf67e0d2f9273f473752686a1